### PR TITLE
fix: drop and score P2P objects the peer was never asked for

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented

Several P2P object types travel `inv` → `getdata` → object and are never pushed on their own. When such a message arrives that we never asked for, we process it anyway, and for some of them the rejection path costs the sender nothing at all.

The clearest case is `CLSIG`. `ChainlockHandler::ProcessNewChainLock` short-circuits with **no misbehaviour penalty** for any CLSIG at or below our best ChainLock — the height check runs before BLS verification, deliberately, to avoid a verification DoS. Because every distinct signature blob hashes differently, the seen-cache dedup above it never catches a varied blob, so a peer can repeat that work indefinitely for free: churning the (bounded) 1024-entry seen cache and competing with real LLMQ traffic, since `CLSIG` sits on the quorum priority queue (`IsQuorumPriorityMessage`).

`QFCOMMITMENT` has the same shape. Most of the checks in `CQuorumBlockProcessor::ProcessMessage` return without scoring the peer — deliberately, since we may just be lagging behind or on another chain — so an unsolicited sender can drive the block lookups and mineable-commitment probes repeatedly at no cost. The DKG messages are retained in the per-phase pending queues until a worker gets round to verifying their signatures.

The net layer already tracks, per peer, what we asked for, and governance objects and votes already authorise incoming payloads against it. These three object types simply weren't doing so — the underlying `ReceivedResponse` was called and its return value discarded.

## What was done?

Authorise three object types against the request tracker, dropping and scoring anything we have no in-flight GETDATA for.

The existing `PeerConsumeObjectRequest` is too weak for this. It resolves to `TxRequestTracker::ReceivedResponse`, which succeeds for any non-`COMPLETED` announcement — including the `CANDIDATE` created the moment an INV is processed. The transition to `REQUESTED` only happens later, when `SendMessages` schedules the GETDATA. A peer could therefore send `INV(hash)` immediately followed by the payload and authorise itself, which defeats the gate entirely; completing a sole candidate also erases it, so the per-peer announcement cap does not bound the repetition. This is reachable in production, not just in tests: sends for masternode connections are deliberately batched at ~100 ms intervals.

So this adds `TxRequestTracker::ReceivedRequestedResponse`, which succeeds only in the `REQUESTED` state, exposed as `PeerConsumeGetDataResponse`. On failure it leaves the announcement untouched, so a premature payload is rejected without suppressing the legitimate GETDATA that follows. Governance keeps the existing announced-or-requested semantics via `PeerConsumeObjectRequest`, which is unchanged.

The three call sites:

* **`CLSIG`** — `net_processing.cpp`. Authorised after the spork-19 gate, so a CLSIG dropped while ChainLocks are disabled does not burn a later retransmit.
* **`QCONTRIB` / `QCOMPLAINT` / `QJUSTIFICATION` / `QPCOMMITMENT`** — `llmq/net_dkg.cpp`. The check is placed last, after the MNAuth, size and structural checks, so every existing rejection path and its score is unchanged. Replaces the previous `PeerEraseObjectRequest` call, which discarded the answer.
* **`QFCOMMITMENT`** — `llmq/blockprocessor.cpp`. Runs before any of the lookups above — the opposite placement from the DKG site, deliberately. There the pre-existing checks are cheap (MNAuth, size, structure) and gating last preserves their heavier penalties; here the pre-existing checks *are* the expensive part — block-index lookups and mineable-commitment probes under `::cs_main` — so an unsolicited sender must be turned away before reaching them. (The gate takes `::cs_main` too, but only for a tracker probe: a brief, bounded hold rather than chain-state traversal on the sender's behalf.) This knowingly softens the penalty for unsolicited *garbage* from 100 to 10 — such a peer now survives ten messages instead of one — but each of those messages is now a deserialize-hash-drop instead of chain lookups, and anything we actually requested still faces the full existing penalties. `CQuorumBlockProcessor` holds no `PeerManagerInternal` reference (`net_processing` already includes this header, so taking one would make the dependency circular), so the check is passed in as a predicate from the call site. Replaces the `m_to_erase` round-trip, which resolved to the weaker `ReceivedResponse`.

A shared `UNREQUESTED_OBJECT_MISBEHAVIOR_SCORE{10}` lives in `msg_result.h`, matching the existing "moderate" penalties nearby (`mnauth` from an unknown masternode, a DKG message from a non-verified peer, an invalid CLSIG).

### Late and superseded answers are not misbehaviour

(The four commits addressing this are authored by UdjinM6, from review.)

A strict in-flight-only gate conflates an unsolicited push with an honest answer to a GETDATA of ours whose tracker entry is simply gone. Two routine paths remove it: **expiry** — a REQUESTED announcement past `GetObjectInterval()` (only 5s for `MSG_CLSIG`) is completed, and deleted outright when it was the sole announcement for its hash, the common case; and **`ForgetTxHash`** — accepting the object from any source (a ChainLock we signed ourselves, or one submitted over RPC) erases every peer's announcement, stranding an in-flight request. Neither is misbehaviour, but both would have been scored 10 — and since the score does not decay within a connection, a run of them discourages the peer's address. On the long-lived masternode connections these object types travel over, that is the wrong outcome.

So `SendMessages` additionally records, per peer, the GETDATA-only objects we actually sent a GETDATA for, and `PeerConsumeGetDataResponse` now reports `REQUESTED`, `LATE` or `UNREQUESTED`; only `UNREQUESTED` is scored, and `LATE` is processed as before. Only we ever write to that record, so a peer still cannot authorise its own payload by announcing it first. The record is bounded on three axes, each closing an abuse: **consumption** — an entry is erased by the answer it authorises, so one GETDATA buys exactly one accepted object and the induced payload cannot be replayed; **time** — entries age out after two request intervals of the requested type, so an unanswered request cannot be banked and redeemed later; **type** — both the match and the window come from the type *we requested*, never from the answer, so announcing a hash as `MSG_CLSIG` and answering with a DKG message is not authorised. The 256-entry per-peer LRU cap is only a memory ceiling: early eviction costs the grace for the oldest requests, never correctness.

The gated types are now named once in `IsGetDataOnlyObject`, together with why the other inv-driven types are excluded.

### What the gate does not close

For the DKG messages and `QFCOMMITMENT` the gate is the substance of the fix: those payloads were retained in pending queues or driving block lookups before any cheap check could reject them, and requiring a GETDATA bounds that work outright.

For `CLSIG` the picture is narrower than the motivation above may suggest. The gate closes the *push* path, but a peer can still get itself solicited: announce the hash, and on non-masternode connections `SendMessages` runs after every processed message, so the GETDATA goes out almost immediately (masternode connections batch sends at ~100 ms, same outcome slightly later). A stale solicited CLSIG — at or below our best ChainLock — is then still processed and still costs the sender nothing, since `ProcessNewChainLock`'s height check deliberately exits without penalty. So for the stale-CLSIG churn scenario specifically, the gate adds one INV and one GETDATA round-trip per blob rather than eliminating the free work; what it does guarantee is that nothing reaches the handler without our request, and that each request pays for exactly one payload. Closing the remainder means scoring stale *solicited* CLSIGs or rate-limiting per-peer CLSIG requests — both touch honest relay timing (a peer lagging behind us is normal) and are left as a follow-up.

### Deliberately not gated

`PLATFORMBAN` looks like it belongs in this set — it is only ever pushed from `ProcessGetData` in-tree — but it has **no local ingress**: no RPC, no internal producer. The originating Dash Platform node injects a ban by pushing the P2P message straight to a Dash Core peer, so the first hop is always unsolicited by design. `p2p_platform_ban.py` covers this. A comment at the handler records the reasoning so the omission does not read as an oversight.

The remaining inv-driven types were considered and left alone because each has a legitimate unsolicited-push path that would need a carve-out rather than a plain gate: `QSIGREC` (proactive relay to peers that sent `QSENDRECSIGS`), `MSG_DSQ` (`SENDDSQUEUE` / `WantsDSQ::ALL`), `ISDLOCK` (pushed alongside `MERKLEBLOCK` for BIP37 clients), and `SPORK` (bulk push in reply to `GETSPORKS`). `MSG_TX` / `MSG_DSTX` keep upstream Bitcoin semantics.

## How Has This Been Tested?

Built with autotools on macOS (aarch64-apple-darwin).

New coverage:

* `src/test/llmq_chainlock_tests.cpp` — `unrequested_clsig_is_dropped_and_scored`. An unsolicited peer's CLSIG never reaches `ProcessNewChainLock`, asserted via the handler's `AlreadyHave` (i.e. the hash was never recorded in the seen cache). It is sent twice and charged for both: without the gate the second copy would hit the seen-cache dedup and cost nothing, so this is what distinguishes the gate from the pre-existing invalid-CLSIG penalty. A peer whose GETDATA is in flight gets through, and its score is asserted exactly, so the gate charging an authorised peer as well would fail the test.
* `src/test/net_tests.cpp` — `peer_getdata_response_requires_an_inflight_request`. A bare announcement is rejected and leaves the candidate intact; the same inv authorises exactly once after `SendMessages` has issued the GETDATA.
* `src/test/llmq_chainlock_tests.cpp` also covers the INV-then-payload race directly: announcing and immediately pushing the CLSIG is scored, and the identical payload is only accepted after `SendMessages` runs. It also gains an end-to-end replay: the same CLSIG, accepted once, is scored when sent again.
* `test/functional/feature_llmq_dkg_intake.py` — a well-formed `QCONTRIB` that passes every earlier check (verified sender, known quorum, size, structure) and is rejected purely for being unrequested.
* `src/test/net_tests.cpp` — four cases pinning the late-answer grace and its bounds: the expiry and `ForgetTxHash` paths each yield `LATE` for the first answer (asserting the tracker record is gone entirely, not merely COMPLETED) and `UNREQUESTED` for a second copy; a hash announced as `MSG_CLSIG` and answered as `MSG_QUORUM_CONTRIB` inside the window is not authorised; and an answer one second past the two-interval boundary is `UNREQUESTED`. The boundary cases are expressed in the CLSIG request-interval constants so they sit exactly on the edge — relaxing `<=` to `<` fails them.
* `src/test/fuzz/txrequest.cpp` — `ReceivedRequestedResponse` is modelled in the harness as a new command, asserting against the naive model that the call completes an announcement precisely when it was REQUESTED and leaves everything else, notably an INV-created CANDIDATE, untouched.

Both unit tests were mutation-checked: disabling the CLSIG gate fails 3 assertions, and relaxing `ReceivedRequestedResponse` back to the `ReceivedResponse` semantics fails 4.

Full `make check` passes. Functional tests run: `feature_llmq_chainlocks.py`, `feature_llmq_signing.py` (both variants), `feature_llmq_rotation.py`, `feature_llmq_is_cl_conflicts.py`, `feature_llmq_dkgerrors.py`, `feature_llmq_dkg_intake.py`, `rpc_verifychainlock.py`, `p2p_platform_ban.py` — all pass. `lint-circular-dependencies.py`, `lint-python.py` and `lint-whitespace.py` are clean.

Re-verified at the current head, i.e. with the four review follow-up commits included: full `make check`, `feature_llmq_chainlocks.py`, `feature_llmq_dkg_intake.py` and `p2p_platform_ban.py` all pass, the fuzz harness TU compiles, and the three lints above are clean.

`p2p_platform_ban.py` is what caught the `PLATFORMBAN` case: it was gated in an earlier revision of this branch and the test failed, which is what prompted checking for a local ingress.

## Breaking Changes

None for peers running Dash Core. All three message types have been inv/getdata-only since they were introduced (`CLSIG` since "Implement and enforce ChainLocks", 2019), so no released version pushes them unsolicited.

Third-party software that pushes these messages rather than waiting for our GETDATA will now be scored 10 per message and disconnected after ten of them. No such producer exists for these three types — that property is exactly what separates them from `PLATFORMBAN`.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned this pull request to a milestone

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by thepastaclaw** [comment]
✅ Final review complete — no blockers (commit 5c35e967)

File: src/net_processing.cpp [hunk 1]
```diff
@@ -5537,16 +5541,35 @@ void PeerManagerImpl::ProcessMessage(
...
-                WITH_LOCK(::cs_main, m_object_request.ReceivedResponse(pfrom.GetId(), CInv{MSG_CLSIG, hash}));
-                PostProcessMessage(m_clhandler.ProcessNewChainLock(pfrom.GetId(), clsig, *m_llmq_ctx->qman, hash), pfrom.GetId());
+                const CInv clsig_inv{MSG_CLSIG, ::SerializeHash(clsig)};
+                // A CLSIG is only ever sent in reply to a GETDATA (see ProcessGetData), so one with
+                // no matching announcement was never asked for. Drop it before ProcessNewChainLock,
+                // which exits without any penalty for a CLSIG at or below our best ChainLock -- and
+                // since every distinct signature blob hashes differently, an unsolicited peer could
+                // otherwise repeat that free work indefinitely. Consume after the spork gate so a
+                // CLSIG dropped while ChainLocks are disabled does not burn a later retransmit.
+                if (!WITH_LOCK(::cs_main, return PeerConsumeObjectRequest(pfrom.GetId(), clsig_inv))) {
```
Comment (thepastaclaw) [inline src/net_processing.cpp]: **🔴 Blocking: An INV announcement bypasses all three unsolicited-object gates**

`PeerConsumeObjectRequest()` delegates to `TxRequestTracker::ReceivedResponse()`, which returns true for every non-`COMPLETED` entry, including the `CANDIDATE` created immediately when an INV is processed. The transition to `REQUESTED` only occurs later when `SendMessages()` schedules GETDATA. A peer can therefore send `INV(hash)` followed by the matching payload before GETDATA and pass the gate. The CLSIG test demonstrates exactly this path: it calls `AnnounceInv()` and then sends the CLSIG without calling `SendMessages()`, yet expects the object to reach `ProcessNewChainLock()`. This is reachable through the production scheduler as well because sends for masternode connections are intentionally skipped between most message-processing iterations and emitted only every 100 ms. The same predicate is used by the DKG gate at `src/llmq/net_dkg.cpp:490` and the QFCOMMITMENT gate at `src/llmq/blockprocessor.cpp:80`. Completing a sole candidate removes it from the tracker, so the per-peer announcement cap does not prevent repeating INV-plus-payload pairs. Add a distinct tracker operation that succeeds only for `REQUESTED` entries and use it for these GETDATA-only object types, while preserving the existing announcement-or-request semantics for governance. Tests should reject an INV immediately followed by a payload and accept it only after `SendMessages()` has issued GETDATA.

<sub>source: ['codex']</sub>

File: src/msg_result.h [hunk 2]
```diff
@@ -14,6 +14,19 @@
...
+/**
+ * Misbehaviour score for an object message that the peer was never asked for.
+ *
+ * Several object types (CLSIG, QFCOMMITMENT, the DKG messages, PLATFORMBAN) are only ever sent in
+ * reply to a GETDATA, so a message with no matching announcement is unsolicited by definition and
+ * is dropped before it can cost us any work. The score is deliberately tiny because the per-object
+ * request expires after GetObjectInterval() -- as little as 5s for MSG_CLSIG -- after which an
+ * honest peer answering our GETDATA late is indistinguishable from one that was never asked. At
+ * this score a peer is only discouraged after DISCOURAGEMENT_THRESHOLD such messages, which bounds
+ * the abuse a single connection can sustain without penalising ordinary lag.
```
Comment (thepastaclaw) [inline src/msg_result.h]: **💬 Nitpick: Remove PLATFORMBAN from the GETDATA-only score documentation**

This comment says PLATFORMBAN is only sent in response to GETDATA and is dropped when unsolicited, but `ProcessPlatformBanMessage()` explicitly documents that its originating first hop is unsolicited by design and deliberately has no solicitation gate. The constant is not used for PLATFORMBAN. Document that carve-out here so future maintenance does not incorrectly add a gate that rejects legitimate platform ban ingress.

```suggestion
 * CLSIG, QFCOMMITMENT, and the DKG messages are expected only in reply to GETDATA and are
 * dropped when their solicitation check fails. PLATFORMBAN is excluded because its originating
 * first hop is unsolicited by design; see ProcessPlatformBanMessage(). The score is deliberately
```

<sub>source: ['coderabbit']</sub>

**Review by thepastaclaw — CHANGES_REQUESTED** [review]
## Preliminary review — Codex only

The three new solicitation gates use a tracker operation that accepts an INV-created candidate before any GETDATA is sent, so an attacker can still reach the protected CLSIG, DKG, and quorum-commitment processing paths without being scored. The current CLSIG test explicitly exercises this bypass. The PR also contains a contradictory PLATFORMBAN comment and a corrective commit that should be folded into the commits it amends.

Validated blockers were found in the Codex precheck. Sonnet is deferred until a fresh Codex revalidation clears the blocker gate.

### Review provenance
- Codex reviewers: `gpt-5.6-sol` — general (completed), `gpt-5.6-sol` — dash-core-commit-history (completed)
- Verifier: `gpt-5.6-sol` — verifier
- Sonnet: **not run (deferred by blocker gate)**

🔴 1 blocking | 🟡 1 suggestion(s) | 💬 1 nitpick(s)

_1 additional finding(s) omitted (not in diff)._

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/net_processing.cpp`:
- [BLOCKING] src/net_processing.cpp:5564: An INV announcement bypasses all three unsolicited-object gates
  `PeerConsumeObjectRequest()` delegates to `TxRequestTracker::ReceivedResponse()`, which returns true for every non-`COMPLETED` entry, including the `CANDIDATE` created immediately when an INV is processed. The transition to `REQUESTED` only occurs later when `SendMessages()` schedules GETDATA. A peer can therefore send `INV(hash)` followed by the matching payload before GETDATA and pass the gate. The CLSIG test demonstrates exactly this path: it calls `AnnounceInv()` and then sends the CLSIG without calling `SendMessages()`, yet expects the object to reach `ProcessNewChainLock()`. This is reachable through the production scheduler as well because sends for masternode connections are intentionally skipped between most message-processing iterations and emitted only every 100 ms. The same predicate is used by the DKG gate at `src/llmq/net_dkg.cpp:490` and the QFCOMMITMENT gate at `src/llmq/blockprocessor.cpp:80`. Completing a sole candidate removes it from the tracker, so the per-peer announcement cap does not prevent repeating INV-plus-payload pairs. Add a distinct tracker operation that succeeds only for `REQUESTED` entries and use it for these GETDATA-only object types, while preserving the existing announcement-or-request semantics for governance. Tests should reject an INV immediately followed by a payload and accept it only after `SendMessages()` has issued GETDATA.

In `<commit:0ad2759288e>`:
- [SUGGESTION] <commit:0ad2759288e>:1: Fold the corrective follow-up into its originating commits
  Commit `0ad2759288e` corrects the CLSIG test introduced by `b8977068872` and adds locking-contract documentation for the callback introduced by `40353a8ccbd`. These are amendments to two separate logical changes rather than one standalone change. Fold the test hunk into the CLSIG commit and the header hunk into the quorum-commitment commit so each originating commit is complete and the final history does not retain review-fix noise.
```
</details>

File: src/llmq/blockprocessor.cpp [hunk 3]
```diff
@@ -72,8 +73,20 @@ MessageProcessingResult CQuorumBlockProcessor::ProcessMessage(const CNode& peer,
...
+    // A QFCOMMITMENT is only ever sent in reply to a GETDATA (see ProcessGetData), so one we have no
+    // in-flight request for was never asked for. Drop it up front: most of the checks below reject
+    // without scoring the peer -- deliberately, since we may just be lagging behind -- so an
+    // unsolicited peer could otherwise repeat the block lookups and map probes indefinitely. A bare
+    // announcement deliberately does not qualify: it would let the peer authorise its own payload by
+    // sending INV first.
+    if (!consume_request(CInv{MSG_QUORUM_FINAL_COMMITMENT, ::SerializeHash(qc)})) {
+        LogPrint(BCLog::LLMQ, "CQuorumBlockProcessor::%s -- unrequested commitment from peer=%d\n", __func__,
+                 peer.GetId());
+        return MisbehavingError{UNREQUESTED_OBJECT_MISBEHAVIOR_SCORE, "unrequested quorum commitment"};
```
Comment (thepastaclaw) [inline src/llmq/blockprocessor.cpp:85]: **🟡 Suggestion: No dedicated test exercises the new QFCOMMITMENT solicitation gate**

The CLSIG gate got a focused unit test (unrequested_clsig_is_dropped_and_scored, including the INV-then-payload race) and the DKG gate got both a unit test (peer_getdata_response_requires_an_inflight_request) and a functional test (test_unrequested_rejected in feature_llmq_dkg_intake.py). The third call site — CQuorumBlockProcessor::ProcessMessage's new consume_request check at blockprocessor.cpp:82 — has no equivalent. A search of src/test/ and test/functional/ confirms no test constructs a CQuorumBlockProcessor and calls ProcessMessage with an unrequested QFCOMMITMENT, and no functional test sends a raw QFCOMMITMENT the node never asked for. net_tests.cpp's peer_getdata_response_requires_an_inflight_request exercises the shared tracker primitive via MSG_SPORK, which gives confidence in the primitive itself but not that the predicate is wired correctly at this call site (e.g. firing before rather than after the block/mineable-commitment lookups it's meant to gate, as the commit message claims). Given this touches consensus-adjacent llmq code, a small addition — a synthetic unrequested CFinalCommitment with a predicate returning false, asserting the MisbehavingError and that no lookups occurred — would close this gap.

<sub>source: ['claude']</sub>

**Review by thepastaclaw — COMMENTED** [review]
## Final validation — Codex + Sonnet

This PR fixes a real DoS gap: three GETDATA-only P2P object types (CLSIG, QFCOMMITMENT, DKG push messages) were previously processed without confirming an in-flight request existed. The current head resolves the prior-review blocker by introducing TxRequestTracker::ReceivedRequestedResponse, which strictly requires State::REQUESTED and correctly rejects a bare INV-created CANDIDATE, closing the INV-then-payload bypass confirmed present at the prior reviewed SHA (d01d5f02). Two new Sonnet-only findings remain in scope: the QFCOMMITMENT gate lacks a dedicated regression test unlike its CLSIG/DKG siblings, and net_dkg.cpp uses UNREQUESTED_OBJECT_MISBEHAVIOR_SCORE via a transitive include rather than a direct one.

### Review provenance
- Codex reviewers: `gpt-5.6-sol` — general (completed), `gpt-5.6-sol` — dash-core-commit-history (completed)
- Verifier: `claude-sonnet-5` — final-verifier
- Sonnet reviewers: `claude-sonnet-5` — general (completed), `claude-sonnet-5` — dash-core-commit-history (completed)

🟡 1 suggestion(s) | 💬 1 nitpick(s)

_1 additional finding(s) omitted (not in diff)._

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/llmq/blockprocessor.cpp`:
- [SUGGESTION] src/llmq/blockprocessor.cpp:74-85: No dedicated test exercises the new QFCOMMITMENT solicitation gate
  The CLSIG gate got a focused unit test (unrequested_clsig_is_dropped_and_scored, including the INV-then-payload race) and the DKG gate got both a unit test (peer_getdata_response_requires_an_inflight_request) and a functional test (test_unrequested_rejected in feature_llmq_dkg_intake.py). The third call site — CQuorumBlockProcessor::ProcessMessage's new consume_request check at blockprocessor.cpp:82 — has no equivalent. A search of src/test/ and test/functional/ confirms no test constructs a CQuorumBlockProcessor and calls ProcessMessage with an unrequested QFCOMMITMENT, and no functional test sends a raw QFCOMMITMENT the node never asked for. net_tests.cpp's peer_getdata_response_requires_an_inflight_request exercises the shared tracker primitive via MSG_SPORK, which gives confidence in the primitive itself but not that the predicate is wired correctly at this call site (e.g. firing before rather than after the block/mineable-commitment lookups it's meant to gate, as the commit message claims). Given this touches consensus-adjacent llmq code, a small addition — a synthetic unrequested CFinalCommitment with a predicate returning false, asserting the MisbehavingError and that no lookups occurred — would close this gap.
```
</details>

File: src/msg_result.h (hunk 2 above)
Comment (thepastaclaw) [reply src/msg_result.h]: Resolved in this update — _Remove PLATFORMBAN from the GETDATA-only score documentation_ no longer present.

<sub>Auto-resolved by the review system based on the latest commit diff. If you believe this was closed in error, reopen the thread.</sub>

File: src/net_processing.cpp (hunk 1 above)
Comment (thepastaclaw) [reply src/net_processing.cpp]: Resolved in this update — _An INV announcement bypasses all three unsolicited-object gates_ no longer present.

<sub>Auto-resolved by the review system based on the latest commit diff. If you believe this was closed in error, reopen the thread.</sub>

**Comment by UdjinM6** [comment]
pls see https://github.com/UdjinM6/dash/commits/claude/pr7484-followup/

**Comment by PastaPastaPasta** [comment]
> pls see https://github.com/UdjinM6/dash/commits/claude/pr7484-followup/

Good catch!

**Review by thepastaclaw — COMMENTED** [review]
## Final validation — Codex + Sonnet

Both carried-forward findings are fixed at the current head: the QFCOMMITMENT predicate now has focused end-to-end coverage, and `net_dkg.cpp` directly includes the score's defining header. In the latest delta, the late/superseded GETDATA grace is otherwise sound and well tested, but the documented 256-entry per-peer memory ceiling can retain 512 entries because the LRU cache uses its default truncation threshold.

### Review provenance
- Codex reviewers: `gpt-5.6-sol` — general (completed), `gpt-5.6-sol` — dash-core-commit-history (completed)
- Verifier: `gpt-5.6-sol` — final-verifier (fallback)
- Sonnet reviewers: `claude-sonnet-5` — general (failed), `claude-sonnet-5` — dash-core-commit-history (failed), `claude-sonnet-5` — general (completed), `claude-sonnet-5` — dash-core-commit-history (completed)

💬 1 nitpick(s)

File: src/net_processing.cpp [hunk 4]
```diff
@@ -516,6 +547,17 @@ struct CNodeState {
...
+    //! The GETDATA-only objects (see IsGetDataOnlyObject) we have recently asked this peer for. The
+    //! tracker cannot answer "did we ever ask?" on its own: an announcement is erased once it
+    //! expires as the sole one for its hash, or once the object is accepted from anywhere.
+    //!
+    //! Only we ever add to this, so a peer cannot use it to authorise its own payload. An entry is
+    //! erased by the answer it authorises and ages out after RECENT_OBJECT_REQUEST_TTL_INTERVALS, so
+    //! one GETDATA buys exactly one accepted object, within a bounded window: a peer can neither
+    //! replay the payload it induced a request for, nor bank an unanswered request to spend later.
+    unordered_lru_cache<uint256, RequestedObject, StaticSaltedHasher, MAX_RECENT_OBJECT_REQUESTS>
+        m_recent_object_requests;
```
Comment (thepastaclaw) [inline src/net_processing.cpp:559]: **💬 Nitpick: The cache can retain twice the documented ceiling**

This instantiation supplies `MAX_RECENT_OBJECT_REQUESTS` only as `MaxSize` and leaves `TruncateThreshold` at zero. The `unordered_lru_cache` constructor converts that default to `2 * maxSize`, and `truncate_if_needed()` does not prune while the size is at most that threshold. The cache can therefore retain 512 entries between operations and only truncates back to 256 when the 513th entry is inserted, while the nearby comment and PR description call 256 the memory ceiling. Pass an explicit 256-entry truncation threshold if a hard retained-entry bound is intended, or document 256 as the post-truncation target and 512 as the maximum retained count.

<sub>source: ['claude', 'codex']</sub>

File: src/llmq/blockprocessor.cpp (hunk 3 above)
Comment (thepastaclaw) [reply src/llmq/blockprocessor.cpp:85]: Resolved in `8f7c7db` — _No dedicated test exercises the new QFCOMMITMENT solicitation gate_ no longer present.

<sub>Auto-resolved by the review system based on the latest commit diff. If you believe this was closed in error, reopen the thread.</sub>

**Review by UdjinM6 — APPROVED** [review]
utACK 5c35e96734d9bb217a1a401cb415e7c52ed5c37b
